packages/vm/README.md

```diff
@@ -71,6 +71,20 @@ cd packages/vm
 cargo bench --no-default-features --features cranelift
 ```
 
+## Tools
+
+`module_size` and `module_size.sh`
+
+Memory profiling of compiled modules. `module_size.sh` executes `module_size`, and
+uses valgrind's memory profiling tool (massif) to compute the amount of heap memory
+used by a compiled module.
+
+```
+cd packages/vm
+cargo build
```

Comment:
I tried it, and can confirm that the (global) memory footprint / layout stays the same.

It's tricky to do detailed memory profiling of release builds, because without symbols it's more difficult to identify the relevant parts. In `module_size.sh` we are relying on the function name / source file line number to do the aggregation, and that info is lost for release builds.

I can try to do a detailed analysis of a release build, but so far everything indicates the memory footprint will stay the same.

Reply:
Good to know, thanks.

You can enable debug symbols for release builds as well: https://stackoverflow.com/questions/38803760/how-to-get-a-release-build-with-debugging-information-when-using-cargo. The second answer looks good because it does not change the project's default release profile.

Reply:
Nice. Will give it a shot.

Reply:
This works great, and the resulting binary runs much faster. It produces exactly the same results for memory footprint.